README.md

```diff
@@ -62,42 +62,122 @@ Requirements:
 
 - Python >= 3.8
 - pip >= 19.3.1
-- (Optional) A Python virtual environment (for instance [virtualenv](https://docs.python.org/3.10/tutorial/venv.html) or [miniconda](https://docs.conda.io/en/latest/miniconda.html))
 - [Rust and Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html). If you already have the Rust toolchain installed, make sure it is up-to-date with:
 
    ```bash
    rustup update
    ```
 
-Steps:
+### Recommended: Build with Virtual Environment
 
-- Install developments requirements:
+Using a virtual environment is **strongly recommended** to avoid Python version conflicts and dependency issues.
+
+1. Create and activate a virtual environment:
+
+   ```bash
+   python3 -m venv .venv
+   source .venv/bin/activate  # On Windows: .venv\Scripts\activate
+   ```
+
+2. Install development requirements:
 
    ```bash
    pip install -r requirements-dev.txt
    ```
 
-- Ensure your system can find the building tool `maturin` (installed by previous step).
-  For example, it is placed at _$HOME/.local/bin/maturin_ by default on Ubuntu 20.04.
+3. Build and install in development mode:
+
+   ```bash
+   maturin develop --release
+   ```
+
+4. Run examples:
+
+   ```bash
+   python examples/z_info.py
+   ```
+
+When you're done, deactivate the virtual environment:
+
+```bash
+deactivate
+```
+
+### Alternative: Build without Virtual Environment
+
+If you cannot use a virtual environment, follow these steps carefully:
+
+1. Install development requirements:
+
+   ```bash
+   pip install -r requirements-dev.txt
```

Comment:
On recent python version, [`--break-system-packages`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-break-system-packages) is required outside of virtual environment

Reply:
Thank you, will correct